### ISSUE TITLE
Wrong transformation for transformPatternsTrivialPredicates when DISTINCT predicate operand is NULL

### ISSUE THREAD

**Issue by lukaseder** [jOOQ/jOOQ#14163]
The `QOM.IsDistinctFrom` and `QOM.IsNotDistinctFrom` predicates extend `CompareCondition`, which is the only check done for `Settings.transformPatternsTrivialPredicates` to decide whether a condition (e.g. `a = null`) is reduced to a `nullCondition()`.

This means the following wrong transformation is made:

```sql
-- Input
SELECT a IS DISTINCT FROM NULL, a IS NOT DISTINCT FROM NULL

-- Output
SELECT NULL, NULL
```

**Comment by lukaseder** [comment]
The fix is implemented as part of a new pattern `transformPatternsDistinctFromNull` from here: https://github.com/jOOQ/jOOQ/issues/13593

**Comment by lukaseder** [comment]
Fixed in jOOQ 3.18.0 and 3.17.5 (https://github.com/jOOQ/jOOQ/issues/14164)
